packages/mds-cache/index.ts

```diff
@@ -252,15 +252,14 @@ async function readDevicesStatus(query: { since?: number; skip?: number; take?:
   return values.filter((item: any) => item.telemetry)
 }
 
-// get the provider for a device
-async function getProviderId(device_id: UUID) {
-  return (await readDevice(device_id)).provider_id
-}
-
 // initial set of stats are super-simple: last-written values for device, event, and telemetry
-async function updateProviderStats(suffix: string, device_id: UUID, timestamp: Timestamp | undefined | null) {
+async function updateProviderStats(
```

Comment:
I would probably suggest you rearrange the arguments to make `timestamp` (which is effectively optional) last. Maybe:
```
updateProviderStats(
  suffix: string,
  provider_id: UUID,
  device_id: UUID,
  timestamp: Timestamp | null = null
)
```